compiler/src/iree/compiler/Dialect/Stream/Transforms/EncodeTensors.cpp

```diff
@@ -640,7 +636,14 @@ struct EncodeBindingSubspanOp
     }
 
     // Align the element type, if needed.
-    auto alignedType = alignDispatchTensorType(originalType);
+    IREE::Flow::DispatchTensorType alignedType =
+        alignDispatchTensorType(originalType);
+    if (!alignedType) return op.emitOpError("failed to legalize element type");
+    if (originalType.getAccess() != Flow::TensorAccess::ReadOnly &&
+        needToPackSubByteInterfaceElements(
+            cast<RankedTensorType>(alignedType.getBoundType()))) {
+      return op.emitOpError("unsupported sub-byte dispatch tensor write");
+    }
```

Comment:
why this limitation? this feels like a sharp edge

Reply:
Multiple threads writing to the sub-byte elements belonging to the same byte would require careful synchronization using atomics to avoid racing conditions. I was trying to be conservative to just error out on such cases. But yeah it does make it look a bit weird here. So just removed to let backends to handle such cases. (In SPIR-V it's supported already.)